src/Security/UserChecker.php

```diff
@@ -0,0 +1,324 @@
+            $this->logger->info(
+                sprintf('User %s has been locked for %s minutes', $user->getUsername(), $lockMinutes),
+                ['contao' => new ContaoContext(__METHOD__, ContaoContext::ACCESS)]
+            );
+
+            // Send admin notification
+            $request = $this->requestStack->getCurrentRequest();
+            if ($config->get('adminEmail')) {
+                if ($request && $this->scopeMatcher->isFrontendRequest($request)) {
+                    $realName = sprintf('%s %s', $user->firstname, $user->lastname);
+                } else {
+                    $realName = $user->name;
+                }
+
+                $website = Idna::decode($this->requestStack->getCurrentRequest()->getSchemeAndHttpHost());
+                $lockMinutes = ceil(((int) $config->get('lockPeriod')) / 60);
+
+                $subject = $this->translator->trans('MSC.lockedAccount.0', [], 'contao_default');
+                $body = $this->translator->trans(
+                    'MSC.lockedAccount.1',
+                    [
+                        $user->getUsername(),
+                        $realName,
+                        $website,
+                        $lockMinutes,
+                    ],
+                    'contao_default'
+                );
+
+                $email = new \Swift_Message();
+                $email
+                    ->setTo($config->get('adminEmail'))
+                    ->setSubject($subject)
+                    ->setBody($body, 'text/plain')
+                ;
+
+                $this->mailer->send($email);
+            }
+
+            throw new LockedException(sprintf('This account (%s) has been locked!', $user->getUsername()));
+        }
+    }
+
+    /**
+     * Check whether the account is locked.
+     *
+     * @param User $user
+     */
+    protected function checkIfAccountIsLocked(User $user): void
+    {
+        if (false === $user->isAccountNonLocked()) {
+            $this->setAccountLockedFlashBag($user);
+
+            throw new LockedException(sprintf('This account (%s) has been locked!', $user->getUsername()));
+        }
+    }
+
+    /**
+     * Check whether the account is disabled.
+     *
+     * @param User $user
+     */
+    protected function checkIfAccountIsDisabled(User $user): void
+    {
+        if (false === $user->isEnabled()) {
+            $this->setInvalidLoginFlashBag();
+            $this->logger->info(
+                'The account has been disabled',
+                ['contao' => new ContaoContext(__METHOD__, ContaoContext::ACCESS)]
+            );
+
+            throw new DisabledException(sprintf('This account (%s) has been disabled!', $user->getUsername()));
+        }
+    }
+
+    /**
+     * Check wether login is allowed (front end only).
+     *
+     * @param User $user
+     */
+    protected function checkIfLoginIsAllowed(User $user): void
+    {
+        if ($user instanceof FrontendUser && false === $user->login) {
```

Comment:
Are you sure that the strict type check works here? `$user->login` is a `char(1)` column which is either `''` or `'1'` unless casted to boolean.

Reply:
Good catch! 👍